agent/app/agent_capability.go

```diff
@@ -124,6 +126,14 @@ func (agent *ecsAgent) capabilities() []*ecs.Attribute {
 		capabilities = append(capabilities, taskENIVersionAttribute)
 	}
 
+	if agent.cfg.AWSVPCBlockInstanceMetdata {
```

Comment:
You may want to move this above inside the check for `TaskENIEnabled` since this relies on the `awsvpc` network mode and cni plugins.

Reply:
I thought about and ended up not doing it as it's essentially a nop to do this if Task ENI is not enabled. I'll change it if you insist.

Reply:
Move this above makes the logic more clear to me, let's do it, thanks.